packages/server/src/project/dto/project.dto.ts

```diff
@@ -83,3 +100,27 @@ export class ConfigurableProjectSettings implements Partial<Omit<Project, 'id' |
   @Field({ nullable: true })
   redirectUrl: string | null;
 }
+
+/**
+ * Input type for updating a project's settings
+ */
+@InputType()
+export class ProjectSettingsInput implements Partial<ProjectSettingsModel> {
+  @IsOptional()
+  @IsNotEmpty()
+  @Field()
+  displayProjectName: boolean;
+
+  @IsOptional()
```

Comment:
For `isOptional`, you'll need to update the `@Field` so that it is `@Field({ nullable: true })` and update the type to support that. (I forget if it comes across as undefined or null so it'll be something like `allowSignup?: boolean`